lib/api/filter.rb

```diff
@@ -33,22 +33,19 @@ def initialize(filters, model)
     def parse
       filters.select(&:present?).each do |filter|
         parsed_filter = parse_filter(filter)
-        *associations, attr = parsed_filter[:attr].split(".")
-        if associations.size > 1
+        field = parsed_filter[:field]
+        if field.associations.size > 1
           raise BadRequestError, "Filtering of attributes with more than one association away is not supported"
         end
-        unless virtual_or_physical_attribute?(target_class(model, associations), attr)
-          raise BadRequestError, "attribute #{attr} does not exist"
+        unless field.valid?
+          raise BadRequestError, "attribute #{field} does not exist"
         end
 
-        associations.map! { |assoc| ".#{assoc}" }
-
-        field = "#{model.name}#{associations.join}-#{attr}"
         op    = parsed_filter[:operator]
         expr  = if parsed_filter[:value].kind_of?(Array)
-                  {"OR" => parsed_filter[:value].map { |val| single_expression(field, op, val) }}
+                  {"OR" => parsed_filter[:value].map { |val| single_expression(field.to_s, op, val) }}
```

Comment:
Does single_expression always need a string as the input?  If so, can we do the .to_s within the single_expression method, or could that cause problems?